zero/src/rpc/zeth.rs

```diff
@@ -0,0 +1,35 @@
+use alloy::{providers::Provider, rpc::types::eth::BlockId, transports::Transport};
+use trace_decoder::BlockTrace;
+
+use super::fetch_other_block_data;
+use crate::prover::BlockProverInput;
+use crate::provider::CachedProvider;
+
+pub async fn block_prover_input<ProviderT, TransportT>(
+    cached_provider: std::sync::Arc<CachedProvider<ProviderT, TransportT>>,
+    target_block_id: BlockId,
+    checkpoint_block_number: u64,
+) -> anyhow::Result<BlockProverInput>
+where
+    ProviderT: Provider<TransportT>,
+    TransportT: Transport + Clone,
+{
+    let block_number = match target_block_id {
+        BlockId::Number(block_number) => block_number,
+        _ => return Err(anyhow::anyhow!("block number expected")),
+    };
+
+    let block_trace = cached_provider
+        .get_provider()
+        .await?
+        .raw_request::<_, BlockTrace>("zero_getBlockTraceByNumber".into(), vec![block_number])
```

Comment:
Isn't there a same standard API in `zeth`, `debug_traceBlockByNumber`?

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-api/src/debug.rs#L73

Reply:
zeth is a wrapper over reth. This was done by Francesco earlier. We can name this API whatever we want
https://github.com/0xPolygonZero/zeth/issues/9

Reply:
Ya cool. Created task above to handle this in future